snapcraft/internal/lxd/_errors.py

```diff
@@ -0,0 +1,176 @@
+        "Refer to the documentation at "
+        "https://linuxcontainers.org/lxd/getting-started-cli."
+    )
+
+    def __init__(self, remote):
+        super().__init__(remote=remote)
+
+
+class ContainerLXDSetupError(ContainerError):
+
+    fmt = (
+        "Failed to initialize container: "
+        "Something seems to be wrong with your installation of LXD.\n"
+        "Refer to the documentation at "
+        "https://linuxcontainers.org/lxd/getting-started-cli."
+    )
+
+    def __init__(self):
+        super().__init__()
+
+
+class ContainerStartFailedError(ContainerError):
+
+    fmt = (
+        "Failed to start container: "
+        "The local folder could not be mounted into the container. "
+        "The files /etc/subuid and /etc/subgid need to "
+        "contain this line for mounting the local folder:\n"
+        "    root:1000:1\n"
+        "Note: Add the line to both files, do not "
+        "remove any existing lines.\n"
+        "Restart LXD after making this change.\n"
+    )
+
+    def __init__(self):
+        super().__init__()
+
+
+class ContainerCreationFailedError(ContainerError):
+
+    fmt = (
+        "Failed to create container: "
+        "A new LXD container could not be created.\n"
+        "Refer to the documentation at "
+        "https://linuxcontainers.org/lxd/getting-started-cli."
+    )
+
+    def __init__(self):
+        super().__init__()
+
+
+class ContainerNetworkError(ContainerError):
+
+    fmt = (
+        "Failed to get a network connection in the container: "
+        "Could not successfully ping {url!r}. "
+        "If using a proxy, check its configuration.\n"
+        "Refer to the documentation at "
+        "https://linuxcontainers.org/lxd/getting-started-cli."
+    )
+
+    def __init__(self, url):
+        super().__init__(url=url)
+
+
+class ContainerRunError(ContainerError):
+
+    fmt = (
+        'The following command failed to run: '
+        '{command!r} exited with {exit_code}\n'
+    )
+
+    def __init__(self, *, command, exit_code):
+        if isinstance(command, list):
+            command = ' '.join(command)
+        super().__init__(command=command, exit_code=exit_code)
+
+
+class ContainerSnapcraftCmdError(ContainerRunError):
+
+    fmt = (
+        'Snapcraft command failed in the container: '
+        '{command!r} exited with {exit_code}\n'
+    )
+
+    def __init__(self, *, command, exit_code):
+        if isinstance(command, list):
+            command = ' '.join(command)
+        super().__init__(command=command, exit_code=exit_code)
+
+
+class SnapdQueryError(ContainerError):
+
+    fmt = (
+        "Failed to get snap in container: "
+        "Error querying {snap!r} snap: {message}"
+    )
+
+    def __init__(self, snap, message):
+        super().__init__(snap=snap, message=message)
+
+
+class SnapdConnectionError(ContainerError):
+
+    fmt = (
+        "Failed to get snap in container: "
+        "Could not connect to {api!r}."
```

Comment:
Same here: what does it mean to "get" a snap, and are we sure it's in the container? How about "Failed to get information for snap {snap!r}: could not connect to {api!r}."